src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -0,0 +1,71 @@
+package tech.cryptonomic.conseil.metadata
+
+import tech.cryptonomic.conseil.config.MetadataOverridesConfiguration
+import tech.cryptonomic.conseil.config.Platforms.PlatformsConfiguration
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, Entity, Network, Platform}
+import tech.cryptonomic.conseil.metadata.Data.{AttributeData, EntityData, NetworkData, PlatformData}
+import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
+import tech.cryptonomic.conseil.util.OptionUtil.when
+import tech.cryptonomic.conseil.util.ConfigUtil
+
+import scala.concurrent.Future.successful
+import scala.concurrent.{ExecutionContext, Future}
+
+class MetadataService(config: PlatformsConfiguration,
+                      tezosPlatformDiscoveryOperations: TezosPlatformDiscoveryOperations,
+                      metadataOverrides: MetadataOverridesConfiguration) {
+
+  def getPlatforms: List[Platform] = {
+    ConfigUtil
+      .getPlatforms(config)
+      .flatMap(platform => PlatformData(PlatformPath(platform.name), platform).apply(metadataOverrides))
```

Comment:
I see a general pattern here apparently, e.g. for a given platform `layer`(platform, network, entity, attribute)

`layer => LayerData(pathBuilder(layer.name), layer).apply(metadataOverrides)`

this makes me think that 
1. The "LayerData" is a short-lived storage of data that will be passed to a function call
2. often, if not always, `layer.name` is passed along with the `layer` to the `LayerData`, hence we could pass `layer` only

Given `2`, we can think of converting `LayerData(...).apply(overrides)` to a function of the same name that simply gets all arguments together, e.g.

```scala
object Data {
  def platform(platform: Platform, overrides: OverridesConf): Option[Platform] = {
    val path = PlatformPath(platform.name)
    when(overrides.isVisible(path)) {
      val nameOverride = overrides.platform(path).map(_.displayName)
      nameOverride.fold(platform){
        overridden => platform.copy(displayName = overridden)
      }
    }
  }
```
 evaluate the code above, if it makes sense.

I also approached the code to avoid copying the layer if not needed. The logic might or might not be easier to follow

Reply:
I must provide ``Path`` too, but I refactored it in a similar way.

Reply:
Isn't the `PlatformPath` derived from the platform itself?

Reply:
Yes, it is. But you cannot create a ``Path`` based on an ``Entity`` or an ``Attribute`` so I provide ``Path`` everywhere to be consistent.